src/derivative_utils.jl

```diff
@@ -462,7 +462,9 @@ function calc_W(integrator, cache, dtgamma, repeat_step, W_transform=false)
   if (f isa ODEFunction && islinear(f.f)) || (f isa SplitFunction && islinear(f.f1.f))
```

Comment:
Should a check for `SplitAlgorithm`s be added, similar to the `calc_W!` version?

Reply:
Checking for SplitFunction is probably better.

Reply:
In `calc_W!` (and some other places, if I remember correctly) we check for both